hammett/widgets/calendar_widget.py

```diff
@@ -0,0 +1,781 @@
+"""The module contains the implementation of the calendar widget."""
+
+import calendar
+import itertools
+import json
+from datetime import date, datetime, timezone
+from enum import Enum
+from typing import TYPE_CHECKING
+
+from dateutil.relativedelta import relativedelta
+
+from hammett.core.button import Button
+from hammett.core.constants import DEFAULT_STATE, RenderConfig, SourceTypes
+from hammett.core.handlers import register_button_handler
+from hammett.core.mixins import I18NMixin
+from hammett.utils.translation import gettext as _
+from hammett.widgets.base import BaseWidget
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from telegram import Update
+    from telegram.ext import CallbackContext
+    from telegram.ext._utils.types import BD, BT, CD, UD
+    from typing_extensions import Self
+
+    from hammett.types import Handler, Keyboard, State
+
+
+class Unit(str, Enum):
```

Comment:
it's not only used in this module
 
```suggestion
class CalendarUnit(str, Enum):
```